### PR TITLE
BUG: MultiIndex truncated for some non-list arguments

### LINKED ISSUE #14794: Incorrect sequence handling with MultiIndex.from_tuples
```python
T = tuple(tuple(range(k)) for k in range(1, 8))
#Note: min(map(len, T)) == 1 and max(map(len, T)) == 7
bad = pd.MultiIndex.from_tuples(T)
bad.nlevels  # == 1 ???

good = pd.MultiIndex.from_tuples(list(T))
good.nlevels # == 7
```
#### Problem description
The docstring leaves some ambiguity as to whether a tuple or list can be passed in.  It does say a list of tuples at the top, but in the argument specification, it hints that tuples can be a list/sequence of tuple-likes.  A tuple of tuples is a sequence of tuple-likes.  I find it strange that my tuple of tuples was truncated to a single level MultiIndex, while my list of tuples was not.  
Changing https://github.com/groutr/pandas/blob/master/pandas/indexes/multi.py#L983
to call `itertools.zip_longest` would fix the truncation issue.

I also don't understand why sometimes the arrays are transposed and every other case we don't transpose.
https://github.com/groutr/pandas/blob/master/pandas/indexes/multi.py#L980-L983

#### Expected Output
I would expect lists and tuples and other sequences of tuple-likes (sets, for example) to be treated the same.  Do lists have a special meaning here?

This is in the latest master (588e29d23fa289916a8b057d093af62f35c6dcf6) and pandas 0.19.1.
I'd be happy to submit a PR if this is unintended behavior

### PR DESCRIPTION
**Pull request by groutr**
closes #14794.

WIP.  Discussion/suggestions welcome.

### CODE REVIEW

File: pandas/indexes/multi.py [hunk 1]
```diff
@@ -964,24 +969,18 @@ def from_tuples(cls, tuples, sortorder=None, names=None):
...
 
-            arrays = list(lib.tuples_to_object_array(tuples).T)
-        elif isinstance(tuples, list):
-            arrays = list(lib.to_object_array_tuples(tuples).T)
+        if isinstance(tuples, np.ndarray):
+            arrays = lib.tuples_to_object_array(tuples).T
         else:
-            arrays = lzip(*tuples)
-
+            arrays = zip_longest(*tuples)
```
Comment (jreback) [inline pandas/indexes/multi.py]: revert all except for this change here

**Comment by jreback** [comment]
pls add tests first. just make the simple change and get it to work. then you can do follows to restructure if necessary / more clear. There are a lot of things going going on here (IOW lots of routines hit this), and lots of variations of accepted, so need to be quite careful (and more so about performance).

File: pandas/indexes/multi.py [hunk 2]
```diff
@@ -10,7 +10,7 @@
 import pandas.index as _index
 from pandas.lib import Timestamp
 
-from pandas.compat import range, zip, lrange, lzip, map
```
Comment (groutr) [inline pandas/indexes/multi.py]: zip_longest changed names between Python 2/3 (it was izip_longest in Python 2)

**Comment by groutr** [comment]
Thank you.  I guess I got carried away.  I'll do a fresh start and open a new PR.

**Pull request closed without merging**